kvirt/k3s/__init__.py

```diff
@@ -96,22 +107,38 @@ def create(config, plandir, cluster, overrides):
     if os.path.exists("manifests") and os.path.isdir("manifests"):
         data['files'] = [{"path": "/root/manifests", "currentdir": True, "origin": "manifests"}]
     k = config.k
-    result = config.plan(plan, inputfile='%s/bootstrap.yml' % plandir, overrides=data)
+    bootstrap_overrides = data.copy()
+    bootstrap_install_k3s_args = install_k3s_args.copy()
+    if sdn != 'flannel':
+        bootstrap_install_k3s_args.append("INSTALL_K3S_EXEC='--flannel-backend=none'")
+    bootstrap_install_k3s_args = ' '.join(bootstrap_install_k3s_args)
+    bootstrap_overrides['install_k3s_args'] = bootstrap_install_k3s_args
+    result = config.plan(plan, inputfile='%s/bootstrap.yml' % plandir, overrides=bootstrap_overrides)
     if result['result'] != "success":
         os._exit(1)
+    nodes_overrides = data.copy()
+    nodes_install_k3s_args = install_k3s_args.copy()
+    if sdn != 'flannel':
+        nodes_install_k3s_args.append("INSTALL_K3S_EXEC='--disable-network-policy --no-flannel'")
+    nodes_install_k3s_args = ' '.join(nodes_install_k3s_args)
+    nodes_overrides['install_k3s_args'] = nodes_install_k3s_args
     if masters > 1:
         pprint("Deploying extra masters")
-        if 'name' in data:
-            del data['name']
-        config.plan(plan, inputfile='%s/masters.yml' % plandir, overrides=data)
+        config.plan(plan, inputfile='%s/masters.yml' % plandir, overrides=nodes_overrides)
     firstmasterip, firstmastervmport = _ssh_credentials(k, firstmaster)[1:]
     with open("%s/join.sh" % clusterdir, 'w') as f:
```

Comment:
If someone install K3s without wanting any workers and therefore setting `workers == 0` the `join.sh` file is written to no use. And it's totally valid to have a K3s cluster with no workers, as Pod workloads can execute on masters.
So shouldn't the check for `if workers > 0` be before this and so forth ....

Reply:
no, because one still has the option to scale afterwards, which is simpler because we took care of generating this file (which is really simple to do)

Reply:
I didn't think of that. Of course. Why didn't I think of that - dang!